repos/fdbt-netex-output/src/netex-convertor/period-tickets/periodTicketNetexHelpers.ts

```diff
@@ -313,166 +313,171 @@ const getMultiServiceList = (
+                                        id: `op:${product.productName}@${name}@p-ticket@${
+                                            userPeriodTicket.passengerType
+                                        }@${product.productDuration.replace(' ', '-')}`,
+                                        Amount: { $t: `${product.productPrice}` },
+                                        TimeIntervalRef: {
+                                            version: '1.0',
+                                            ref: `op:Tariff@${product.productName}@${product.productDuration.replace(
+                                                ' ',
+                                                '-',
+                                            )}`,
+                                        },
+                                    },
+                                    ColumnRef: {
+                                        version: '1.0',
+                                        ref: `op:${product.productName}@${name}@p-ticket@${userPeriodTicket.passengerType}`,
+                                    },
+                                    RowRef: {
+                                        version: '1.0',
+                                        ref: `op:${
+                                            product.productName
+                                        }@${name}@p-ticket@${product.productDuration.replace(' ', '-')}`,
                                     },
-                                },
-                                ColumnRef: {
-                                    version: '1.0',
-                                    ref: `op:${product.productName}@${name}@p-ticket@${userPeriodTicket.passengerType}`,
-                                },
-                                RowRef: {
-                                    version: '1.0',
-                                    ref: `op:${product.productName}@${name}@p-ticket@${product.productDuration.replace(
-                                        ' ',
-                                        '-',
-                                    )}`,
                                 },
                             },
                         },
                     },
                 },
             },
-        },
-    }));
+        }));
+    });
 };
 
 const getFlatFareList = (
     userPeriodTicket: FlatFareTicket | SchemeOperatorFlatFareTicket,
     ticketUserConcat: string,
 ): NetexObject[] =>
-    userPeriodTicket.products.map(product => ({
-        version: '1.0',
-        id: `op:${product.productName}`,
-        Name: { $t: `${product.productName}` },
-        includes: {
-            FareTable: {
+    userPeriodTicket.products.flatMap(product => {
+        return product.salesOfferPackages.map(salesOfferPackage => {
+            return {
                 version: '1.0',
-                id: `op:${product.productName}@p-ticket@${userPeriodTicket.passengerType}`,
+                id: `op:${product.productName}`,
```

Comment:
Will this not break if there are multiple sales offer packages under one product?

Reply:
@Aaron-Collis-IW  I've checked with Jake and we agree this will be a breaking change if we leave this as is, because it will create multiple of the same id's if there are multiple SOP's for a product (id's have to be unique).

The proposed fix is to change  id: `op:${product.productName}` to  id: `op:${product.productName}@${salesOfferPackage.name}` which would make them unique. 

Lastly, usually when we change an ID we also have to check to see if there are any ref's which match it, and change them too - in this case we dont have to, as the ID isnt used anywhere.